### PR TITLE
REST API: Services API section

### PR DESCRIPTION
**Pull request by mariodavid**

### CODE REVIEW

File: content/modules/rest/pages/business-logic.adoc [hunk 1]
```diff
@@ -1,20 +1,208 @@
 = Business Logic
 
-== Service Invocation
+When interacting with the REST API it is oftentimes needed to have an application-level layer of business logic in place that represents the Invocation point for the API. It can be used for orchestration, validation, or other tasks that should happen when the API Client interacts with the system. The xref:rest:entities-api.adoc[Entities API] does not allow placing additional orchestration business logic in place. The Client in this case directly interacts with the entity layer.
```
Comment (knstvk) [inline content/modules/rest/pages/business-logic.adoc]: "the Invocation point, the API Client" - we usually don't use capitalization if it's not a proper noun or an object in code.
There are other cases below: "Service, Service Interface". 
Plz remove capitalization in these words, it's better for uniformity and IMHO for readability too.

File: content/modules/rest/pages/business-logic.adoc [hunk 2]
```diff
@@ -1,20 +1,208 @@
...
+To use a Spring bean as part of the Jmix Services API, it needs to fulfill the following criteria:
+
+1. The Spring bean needs to implement an Interface (the Service Interface), that exposes the method signatures.
+2. The Spring bean needs to be registered in the `rest-services.xml` configuration file
+
+=== Service Exposure
+
+To fulfill the first requirement of implementing a Java interface, let's look at the following example:
+
+The Service Interface allows to expose the methods via the REST Services API:
```
Comment (knstvk) [inline content/modules/rest/pages/business-logic.adoc]: "The service interface allows you to expose ..."

File: content/modules/rest/pages/business-logic.adoc [hunk 3]
```diff
@@ -1,20 +1,208 @@
...
+
+[source,java]
+.CalculationService.java
+----
+public interface CalculationService {
+
+    String NAME = "sample_CalculationService"; // <1>
```
Comment (knstvk) [inline content/modules/rest/pages/business-logic.adoc]: We actually don't use constants for bean names in the framework and Studio doesn't generate them in the apps.
So this part should be rewritten.

File: content/modules/rest/pages/business-logic.adoc [hunk 4]
```diff
@@ -1,20 +1,208 @@
...
+
+After the file has been created, and the services have been defined, you need to register the `rest-services.xml` configuration in the `application.properties` of your Jmix application:
+
+[source,properties]
+.application.properties
+----
+jmix.rest.servicesConfig = rest/sample/rest-services.xml
+----
+
+Once the Service is exposed through the Services API, it is possible to invoke the service as an API Client.
```
Comment (knstvk) [inline content/modules/rest/pages/business-logic.adoc]: probably "it is possible to invoke the service **from** an API client."

File: content/modules/rest/pages/business-logic.adoc [hunk 5]
```diff
@@ -1,20 +1,208 @@
...
+A service method may return a result of a simple data type, an entity, an entity collection, or a serializable POJO. In our case an `int` is returned, so the response body contains just a number.
 
 === Invoke a Service (POST)
 
-TIP: https://doc.cuba-platform.com/restapi-7.2/#rest_api_v2_ex_service_post
+REST API allows execution not only of methods that have arguments of simple data types but also of methods with the following arguments:
+
+* Entities
+* Entity Collections
+* serializable POJOs
```
Comment (knstvk) [inline content/modules/rest/pages/business-logic.adoc]: Serializable POJOs

**Review by knstvk — CHANGES_REQUESTED** [review]
General note: what about taking examples from the real code of the app?

File: content/modules/rest/pages/business-logic.adoc (hunk 2 above)
Comment (mariodavid) [reply content/modules/rest/pages/business-logic.adoc]: done

File: content/modules/rest/pages/business-logic.adoc (hunk 1 above)
Comment (mariodavid) [reply content/modules/rest/pages/business-logic.adoc]: done

File: content/modules/rest/pages/business-logic.adoc (hunk 3 above)
Comment (mariodavid) [reply content/modules/rest/pages/business-logic.adoc]: done

File: content/modules/rest/pages/business-logic.adoc (hunk 5 above)
Comment (mariodavid) [reply content/modules/rest/pages/business-logic.adoc]: done

File: content/modules/rest/pages/business-logic.adoc (hunk 4 above)
Comment (mariodavid) [reply content/modules/rest/pages/business-logic.adoc]: done

**Comment by mariodavid** [comment]
@knstvk I made the following changes/additions:

* switched examples to be aligned with the order management example
* added custom controller section
* Added information about `@Service` requirement instead of the interface

I would propose to merge this section if there are no further problems with the content as is. I guess this section can be improved with the information about datatypes and the security behavior of the custom controller if this is under change right now. But I would do this then later.

**Pull request merged**